docker_images/sklearn/app/pipelines/common.py

```diff
@@ -46,3 +59,78 @@ def __init__(self, model_id: str):
         # use column names from the config file if available, to give the data
         # to the model in the right order.
         self.columns = config.get("sklearn", {}).get("columns", None)
+
+    def _get_output(self, inputs: Any) -> Any:
+        raise NotImplementedError(
+            "Implement this method to get the model output (prediction)"
+        )
+
+    def handle_call(self, inputs: Any) -> Any:
```

Comment:
to me it seems that this is actually `__call__`. So why not implement it here, and remove the one line implementation form the child classes, and only have them implement `_get_output`?

Reply:
> So why not implement it here, and remove the one line implementation form the child classes

I tried to explain it in the comment:

```python
    # even though we only delegate the call, we implement this method have the
    # correct docstring and type annotations
```

(just noticed a missing word there)

That's also why I renamed the method in the parent class -- this way, the concrete classes cannot forget to implement `__call__`. If we don't care for the annotations and docstring, I can do the change you suggested.

Reply:
But we don't really care about the typing and docstring here, do we? these classes are never used by end users or other devs.

Reply:
Maybe, I don't know if and how they're used. I just saw that they were there so I assumed they served a purpose. If they don't, I'm happy to remove this indirection. Maybe @Narsil can comment on that.

Reply:
Oh no the scaffolding aims to kickstart new implementation, then you're free to do whatever you want.

Also the parent class should already yell if `__init__` or `__call__` or not implemented.